octue/utils/encoders.py

```diff
@@ -10,5 +12,8 @@ def default(self, obj):
         if hasattr(obj, "serialise"):
             return obj.serialise()
 
+        if isinstance(obj, datetime.datetime):
+            return str(obj)
+
```

Comment:
this is the default method of serializing any object. Shouldn't be required?

Reply:
It wasn't working for some reason - it complains that `datetime`s aren't serialisable.

Reply:
I get this if I remove it:
```
Error
Traceback (most recent call last):
  File "/Users/Marcus1/repos/octue/octue-sdk-python/tests/resources/test_dataset.py", line 385, in test_to_cloud
    dataset.to_cloud(project_name, **location_parameters)
  File "/Users/Marcus1/repos/octue/octue-sdk-python/octue/resources/dataset.py", line 119, in to_cloud
    serialised_dataset = self.serialise()
  File "/Users/Marcus1/repos/octue/octue-sdk-python/octue/mixins/serialisable.py", line 93, in serialise
    string = json.dumps(self_as_primitive, cls=OctueJSONEncoder, sort_keys=True, indent=4, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/json/__init__.py", line 234, in dumps
    return cls(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/json/encoder.py", line 201, in encode
    chunks = list(chunks)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/json/encoder.py", line 431, in _iterencode
    yield from _iterencode_dict(o, _current_indent_level)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/json/encoder.py", line 325, in _iterencode_list
    yield from chunks
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/json/encoder.py", line 438, in _iterencode
    o = _default(o)
  File "/Users/Marcus1/repos/octue/octue-sdk-python/octue/utils/encoders.py", line 13, in default
    return obj.serialise()
  File "/Users/Marcus1/repos/octue/octue-sdk-python/octue/mixins/serialisable.py", line 93, in serialise
    string = json.dumps(self_as_primitive, cls=OctueJSONEncoder, sort_keys=True, indent=4, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/json/__init__.py", line 234, in dumps
    return cls(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/json/encoder.py", line 201, in encode
    chunks = list(chunks)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/json/encoder.py", line 431, in _iterencode
    yield from _iterencode_dict(o, _current_indent_level)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/json/encoder.py", line 438, in _iterencode
    o = _default(o)
  File "/Users/Marcus1/repos/octue/octue-sdk-python/octue/utils/encoders.py", line 19, in default
    return TwinedEncoder.default(self, obj)
  File "/Users/Marcus1/repos/octue/octue-sdk-python/venv/lib/python3.8/site-packages/twined/utils/encoders.py", line 32, in default
    return json.JSONEncoder.default(self, obj)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type datetime is not JSON serializable
```

Reply:
Ok. So, instead of str I think we should be more careful and format it as RFC3339.

Reply:
Good point. I assume we want timezone information in there? How do we get timezone information if users don't provide it?

Reply:
Maybe we should just serialise it as is but in the `isoformat`?

Reply:
> Maybe we should just serialise it as is but in the `isoformat`?

RFC3339 is a more tightly defined isoformat

Reply:
to resolve this: we decided to isoformat() for simplicity, and let the user worry about timezone clusterfucks.